azurerm/internal/services/compute/virtual_machine_scale_set_update.go

```diff
@@ -47,7 +47,9 @@ func (metadata virtualMachineScaleSetUpdateMetaData) performUpdate(ctx context.C
 		upgradeMode := metadata.Existing.VirtualMachineScaleSetProperties.UpgradePolicy.Mode
 
 		if userWantsToRollInstances {
-			if upgradeMode == compute.Automatic {
+			// If the updated image version is not "latest" and upgrade mode is automatic then azure will roll the instances automatically.
+			// Calling upgradeInstancesForAutomaticUpgradePolicy() in this case will cause an error.
+			if upgradeMode == compute.Automatic && *update.VirtualMachineProfile.StorageProfile.ImageReference.Version == "latest" {
```

Comment:
@koikonom heads up this'll need to check case insensitively (and presumably nil-checked?) since Azure returns this in the case the user specified it in:

```suggestion
			isLatest := false
			if update.VirtualMachineProfile.StorageProfile.ImageReference != nil && update.VirtualMachineProfile.StorageProfile.ImageReference.Version != nil {
				isLatest = strings.EqualsFold(update.VirtualMachineProfile.StorageProfile.ImageReference.Version, "latest")
			}
			if upgradeMode == compute.Automatic && isLatest {
```

since it's possible for this to not have an Image Version (for example if it's provisioned from an existing Image [e.g. Azure Resource ID] rather than a Platform Image [e.g. WindowsServer2019])